bokehjs/src/coffee/model.coffee

```diff
@@ -17,6 +21,42 @@ export class Model extends HasProps
       for cb in callbacks
         @listenTo(@, evt, () -> cb.execute(@))
 
+    @listenTo(@, 'change:js_event_callbacks', () -> @_update_event_callbacks)
+
+
+  _process_event : (event) ->
+    # Given an applicable event, execute the associated callback
+    if not event.constructor.event_name
+       logger.warn('Event is still passed as a string: ' + event)
+       return
+    if not event.constructor.applicable_models.some((m) => m == @.type)
```

Comment:
Well, `m` is a string type representing a model. I know it's easier this way, but for the purpose of being able to reason about the code, this and anything else that uses `@type`, must respect model hierarchy. Otherwise someone will be badly surprised with this behavior.